dashboard/modules/reporter/reporter_head.py

```diff
@@ -473,6 +570,32 @@ async def cpu_profile(self, req) -> aiohttp.web.Response:
         else:
             return aiohttp.web.HTTPInternalServerError(text=reply.output)
 
+    @routes.get("/worker/memory_profile")
+    async def memory_profile(self, req) -> aiohttp.web.Response:
+        if "ip" in req.query:
+            reporter_stub = self._stubs[req.query["ip"]]
+        else:
+            reporter_stub = list(self._stubs.values())[0]
+        pid = int(req.query["pid"])
+        duration_s = int(req.query.get("duration", 30))
+
+        # Default not using `--native`, `--leaks` and `--format` for profiling
+        format = req.query.get("format", "flamegraph")
```

Comment:
is there any way to combine this API with tasks and make tasks as a query instead?